src/test/java/com/uid2/shared/attest/AttestationTokenRetrieverTest.java

```diff
@@ -73,40 +74,35 @@ public void attest_succeed_attestationTokenSet(Vertx vertx, VertxTestContext tes
         attestationTokenRetriever.attest();
         Assertions.assertEquals("test_attestation_token", attestationTokenRetriever.getAttestationToken());
         Assertions.assertEquals("appName=appVersion;Component1=Value1;Component2=Value2", attestationTokenRetriever.getAppVersionHeader());
-        verify(this.responseWatcher, times(1)).handle(200);
+        verify(this.responseWatcher, times(1)).handle(Pair.of(200, expectedResponseBody));
         testContext.completeNow();
     }
 
     @Test
-    public void attest_currentTimeAfterTenMinsBeforeAttestationTokenExpiry_expiryCheckCallsAttest(Vertx vertx, VertxTestContext testContext) throws Exception {
+    public void attest_currentTimeAfterTenMinsBeforeAttestationTokenExpiry_expiryCheckCallsAttestFixedIntervalUntilSuccess(Vertx vertx, VertxTestContext testContext) throws Exception {
         attestationTokenRetriever = getAttestationTokenRetriever(vertx);
 
         when(attestationProvider.isReady()).thenReturn(true);
         when(attestationProvider.getAttestationRequest(any())).thenReturn(new byte[1]);
 
         HttpResponse<String> mockHttpResponse = mock(HttpResponse.class);
-        String expectedResponseBody = "{\"body\": {\"attestation_token\": \"test\",\"expiresAt\": \"2023-08-01T00:00:00.111Z\",\"attestation_jwt_optout\": \"\", \"attestation_jwt_core\": \"\"},\"status\": \"success\"}";
-        when(mockHttpResponse.body()).thenReturn(expectedResponseBody);
-        when(mockHttpResponse.statusCode()).thenReturn(200);
-
-        HttpResponse<String> mockHttpResponseSecondAttest = mock(HttpResponse.class);
-        String expectedResponseBodySecondAttest = "{\"body\": {\"attestation_token\": \"test\",\"expiresAt\": \"2023-08-01T00:00:00.111Z\",\"attestation_jwt_optout\": \"\",\"attestation_jwt_core\": \"\"},\"status\": \"success\"}";
-        when(mockHttpResponseSecondAttest.body()).thenReturn(expectedResponseBodySecondAttest);
-        when(mockHttpResponseSecondAttest.statusCode()).thenReturn(300);
+        String expectedResponseBody = "{\"body\": {\"attestation_token\": \"test\",\"expiresAt\": \"2023-08-01T00:00:00.111Z\",\"attestation_jwt_optout\": \"\",\"attestation_jwt_core\": \"\"},\"status\": \"success\"}";
+        when(mockHttpResponse.statusCode()).thenReturn(200, 401, 401, 200);
```

Comment:
We should use 500 as the error code here. If we use 401, it implies the operator will retry on a 401, which it won't, it will shutdown